src/Wordpress/CustomPostType/Booking.php

```diff
@@ -167,130 +180,178 @@ public function saveAdminBookingFields( $post_id, $post = null, $update = null )
-				$isEdit = count( $existingBookings ) === 1 &&
-					array_values( $existingBookings )[0]->getPost()->post_name === $requestedPostname &&
-					array_values( $existingBookings )[0]->getPost()->post_author === get_current_user_id();
-
-				if ( ( ! $isEdit || count( $existingBookings ) > 1 ) && $post_status != 'canceled' ) {
-                    if ($booking) {
-                        $post_status = 'unconfirmed';
-                        set_transient( 'commonsbooking_overlappingBooking_' . $booking->ID, $booking->ID );
-                    } else {
-                        throw new Exception( 'There is already a booking in this timerange.' );
-                    }
-				}
+		if ( $repetitionStart === null || $repetitionEnd === null ) {
+			throw new BookingDeniedException( __( 'Start- and/or end-date is missing.', 'commonsbooking' ) );
+		}
 
-			}
+		/** @var \CommonsBooking\Model\Booking $booking */
+		$booking = \CommonsBooking\Repository\Booking::getByDate(
+			$repetitionStart,
+			$repetitionEnd,
+			$locationId,
+			$itemId
+		);
 
-            // add internal comment if admin edited booking via frontend
-            if ( $booking && $booking->post_author !== '' && intval( $booking->post_author ) !== intval( get_current_user_id() ) ) {
-                $postarr['meta_input']['admin_booking_id'] = get_current_user_id();
-                $internal_comment = esc_html__( 'status changed by admin user via frontend. New status: ', 'commonsbooking' ) . $post_status;
-                $booking->appendToInternalComment( $internal_comment, get_current_user_id() );
-            }
+		$existingBookings =
+			\CommonsBooking\Repository\Booking::getExistingBookings(
+				$itemId,
+				$locationId,
+				$repetitionStart,
+				$repetitionEnd,
+				$booking->ID ?? null,
+			);
 
+		// delete unconfirmed booking if booking process is canceled by user
+		if ( $post_status === 'delete_unconfirmed' && $booking->ID === $post_ID ) {
+			wp_delete_post( $post_ID );
+			throw new BookingDeniedException(
+				__( 'Booking canceled.', 'commonsbooking' ),
+				add_query_arg( 'cb-location', $locationId, get_permalink( get_post( $itemId ) ) )
+			);
+		}
 
-            $postarr['type']                    = sanitize_text_field( $_REQUEST['type'] );
-            $postarr['post_status']             = $post_status;
-            $postarr['post_type']               = self::getPostType();
-            $postarr['post_title']              = esc_html__( 'Booking', 'commonsbooking' );
-            $postarr['meta_input']['comment']   = $comment;
+		// Validate booking -> check if there are no existing bookings in timerange.
+		if ( count( $existingBookings ) > 0 ) {
+			// checks if it's an edit, but ignores exact start/end time
+			$isEdit = count( $existingBookings ) === 1 &&
+						array_values( $existingBookings )[0]->getPost()->post_name === $requestedPostName &&
+						array_values( $existingBookings )[0]->getPost()->post_author === get_current_user_id();
```

Comment:
Wenn es ein Admin-Edit ist, ist `$isEdit` immer false, obwohl es true sein sollte. Da `get_current_user_id` nie gleich des post_author sein kann.

Reply:
@hansmorb ist das vielleicht der Punkt aus den Tests (da wo der Test-Case fehlt).

Reply:
Hui, da bin ich überfragt @chriwen weißt du, was du damit erreichen wolltest?

Reply:
TLDR: Ich würde den Change annehmen, aber die entsprechende Stelle mal überarbeiten. 

Meine Vermutung ist, das der Code ab L292 eine Buchung auf unconfirmed setzen soll, wenn ein User seine Buchung bearbeitet. Aktuell verstehe ich aber nicht, (1) wieso dafür die Bedingung `count( $existingBookings ) > 0` wahr sein muss und (2) wieso Zeile 299 wahr sein muss. Das ist doch ein Zustand der gar nicht eintreten darf. Also zwei Bookings zur selben Zeit. Denn die Buchung aus $booking kann ja nie in $existingBookings enthalten sein. Wenn ich es richtig verstehe.

Meiner Meinung nach gibt es auch nur einen (unwahrscheinlichen) Fall in dem sich der Code nicht wie erwartet verhält: Angenommen wir machen einen Admin-Edit. Es existiert genau eine andere Buchung, der PostName ist gleich dem requestedPostName und der PostAuthor ist aber ungleich unserem user, da wir der Admin sind. Dann ist `isEdit` false (obwohl es true sein müsste, es ist ja ein Edit). Damit wird die Bedingung in L298 wahr ausgewertet. `$booking` ist ebenfalls true (sonst wäre es ja wahrscheinlich auch kein Edit). Unser Booking Status wird damit `unconfirmed`, obwohl es ein Admin Edit ist (der ja ggf. gar nicht den Status ändern will).